core/coreapi/pin.go

```diff
@@ -44,10 +44,6 @@ func (api *PinAPI) Add(ctx context.Context, p path.Path, opts ...caopts.PinAddOp
 		return fmt.Errorf("pin: %s", err)
 	}
 
-	if err := api.provider.Provide(ctx, dagNode.Cid(), true); err != nil {
-		return err
-	}
-
```

Comment:
Previously, at the end of the Pin process, coreapi would trigger a Provide for the root CID (regardless of prividing strategy).

We now provide according to strategy, and this does not need to happen here (happens in the pinner).